exams/factories.py

```diff
@@ -34,6 +35,48 @@ class Meta:
         model = ExamProfile
 
 
+class ExamRunFactory(DjangoModelFactory):
+    """
+    Factory for ExamRun
+    """
+    course = SubFactory(CourseFactory)
+    exam_series_code = factory.Faker('lexify', text="????_MicroMasters")
+    date_first_schedulable = factory.LazyFunction(
+        lambda: FAKE.date_time_this_year(before_now=True, after_now=False, tzinfo=pytz.utc)
+    )
+    date_last_schedulable = factory.LazyFunction(
+        lambda: FAKE.date_time_this_year(before_now=False, after_now=True, tzinfo=pytz.utc)
+    )
+    date_first_eligible = factory.LazyFunction(
+        lambda: FAKE.date_time_this_year(before_now=False, after_now=True, tzinfo=pytz.utc).date()
+    )
+    date_last_eligible = factory.LazyAttribute(
+        lambda exam_run: exam_run.date_first_eligible + timedelta(days=20)
+    )
+    authorized = False
+
+    class Meta:
+        model = ExamRun
+
+    class Params:
+        scheduling_past = factory.Trait(
+            date_first_schedulable=factory.LazyFunction(
+                lambda: FAKE.date_time_this_year(before_now=True, after_now=False, tzinfo=pytz.utc)
+            ),
+            date_last_schedulable=factory.LazyFunction(
+                lambda: FAKE.date_time_this_year(before_now=True, after_now=False, tzinfo=pytz.utc)
```

Comment:
It seems like there's no guarantee that `date_last_schedulable` comes after `date_first_schedulable`. Can you change this to use a `LazyAttribute` similar to below?